Restructure TLS protocols packages

val06 opened the pull request:
Moving all files related to tls protocols (java, gotls, openssl) under `protocols/tls` directory and structuring them in sub-directories.

this includes also the testutils and testdata used for UTs

### Motivation

Better structuring of code in the repository

### Additional Notes


### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes

<!--
* Write here in detail or link to detailed instructions on how this change can
  be tested/QAd/validated, including any environment setup.
-->

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

hush-hush committed:
Improve FX group doc wording

hush-hush committed:
Update docs/components/fx.md

Co-authored-by: Esther Kim <esthercodes2019@gmail.com>

hush-hush committed:
Update docs/components/fx.md

Co-authored-by: Esther Kim <esthercodes2019@gmail.com>

hush-hush committed:
Improve doc following reviews

val06 committed:
refactored tls packages

val06 committed:
refactored tls packages

kangyili committed:
Fix persistent volume type for local volumes (#17643)

fix volume type

val06 committed:
fixed build and invoke files to work with new java dir

val06 committed:
fixed build and invoke files to work with new java dir

TCheruy committed:
[snmp/traps] Collect telemetry for SNMP traps (#17471)

* Collect telemetry for snmp traps

* Send enrichment failed metric only if > 0

* Tag by snmp version

* Fix tests

* Add release note

* Refactor SnmpPacket
Move metrics name to const

* Lint

* Fix copyright

* Update releasenotes/notes/snmp-traps-collect-telemetry-a8dbf3ec35f2e679.yaml

Co-authored-by: Alexandre Yang <alexandre.yang@datadoghq.com>

* Update metric name

* Update listener tests

* Reduce test timeout

---------

Co-authored-by: Alexandre Yang <alexandre.yang@datadoghq.com>

val06 committed:
added agent-usm.jar to the repo

val06 committed:
added agent-usm.jar to the repo

DanielLavie commented on pkg/network/tracer/tracer_usm_linux_test.go:
```diff
@@ -12,6 +12,8 @@ import (
 	"context"
 	"crypto/tls"
 	"fmt"
+	javatestutil "github.com/DataDog/datadog-agent/pkg/network/protocols/tls/java/testutil"
```
Move to correct section

DanielLavie commented on pkg/network/usm/ebpf_javatls.go:
```diff
@@ -10,6 +10,7 @@ package usm
 import (
 	"bytes"
 	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/network/protocols/tls/java"
```
Move to correct section

val06 committed:
fixed gotls_client after moving to network/protocols/tls directory

val06 committed:
fixed gotls_client after moving to network/protocols/tls directory

guyarb committed:
usm: handle the edge case where gotls is enabled but we are failing to prebuilt (#17700)

GoTLS is supported only in runtime compilation and CO-RE, and we check only the configuration, and not the actual loading type.
If we fallback to prebuilt, then we disable unsupported sub programs.

val06 committed:
fixed linting for imports

val06 committed:
fixed linting for imports

jinroh committed:
[CSPM] Make sure we append the host:* tag in our logs (#17681)

guyarb committed:
usm: Change order of wait (#17704)

jinroh committed:
[CSPM] Skip TestProcessInput/Sleeps when no process returned (#17710)

val06 committed:
fixed very amusing bug: when changing the pkg name to non "main" the gotls_client binary is created as non-executable by `go build` command

val06 committed:
fixed very amusing bug: when changing the pkg name to non "main" the gotls_client binary is created as non-executable by `go build` command

val06 committed:
fixed testdata path in java injection test

val06 committed:
fixed testdata path in java injection test

AlexandreYang committed:
[omnibus] Copy SNMP default and user profiles folders (#17631)

ogaca-dd committed:
Use log.Component in comp/forwarder folder (#17396)

* Use log.Component for NewOrchestratorForwarder
It will help to inject log.Component to NewDefaultForwarder

* Inject logger in NewDefaultForwarder

* mend

* Remove pkg/util/log from default_forwarder.go

* Remove pkg/util/log from domain_forwarder.go

* Remove pkg/util/log from worker.go

* Remove pkg/util/log from forwarder_health.go

* Remove pkg/util/log from comp/forwarderdefaultforwarder/internal

* Remove pkg/util/log transaction.go

* Remove pkg/util/log from bloked_endpoints.go

* Remove pkg/util/log from telemetry.go

* Fix compilation issues in some test files

* Fix missing log.Component

* Fix windows build

pducolin committed:
[e2e] add comment in go.mod to bump test image when bumping test-infra (#17665)

* [e2e] add comment in go.mod to bump test image when bumping test-infra dependency

* [e2e] improve bump comment with instructions to docker image name location

* [e2e] add instructions on how to find the docker image path from the commit hash

usamasaqib committed:
Destroy stack without needing to deploy first (#17685)

* Destroy stack without needing to deploy first

* use shorthand for initialization

paulcacheux committed:
[CWS] skip ad rate limiter test in CI (#17717)

* skip `TestActivityDumpsLoadControllerRateLimiter`

* skip only in CI

paulcacheux committed:
[CWS] block when deleting module in tests (#17720)

paulcacheux committed:
bump `actions/labeler` to v4 (#17721)

AlexandreYang committed:
[corechecks/snmp] Add User Profiles support (#17395)

pducolin committed:
[e2e] workshop solutions (#17546)

* [e2e] workshop solutions

* 🚚 [e2e] move workshop solutions to examples

* [e2e] rename workshop examples

AlexandreYang committed:
[corechecks/snmp] Allow extending default profile with same name (#17632)

val06 committed:
reverted back to gotls_client to compat with kitchen tests

val06 committed:
reverted back to gotls_client to compat with kitchen tests

safchain committed:
[CWS] move event stream implementations to a dedicated package (#17661)

* [CWS] refactor eventstream

* [CWS] refactor event stream

ogaca-dd committed:
E2E tests usage improvement  (#17694)

* Rename scenarios/aws/vm/ec2VM to scenarios/aws/vm/ec2vm

* Update import path for ec2params

* Remove generic type from WithStackName

* Rename suiteOptions to params and move it to its own package

* Update import path for agentparams

* Use explicit generic type for e2e.Run to produce better error message.

* Add import in godoc examples

* Add doc for EC2VMStackDef and for AgentStackDef

* Add WithDevMode documentation

* Update test-infra-definition

* Run inv tidy-all

keisku committed:
Change log level to debug when failing to get gce tags (#17701)

* change log level to debug when failing to get gce tags

* use fmt.Errorf instead of debug log

julien-lebot committed:
Replace iwr with Systen.Net.WebRequest (#17707)

Also print the content of the NuSpec file for troubleshooting

safchain committed:
[CWS] increase ctime in test (#17737)

safchain committed:
[CWS] increase syscalls map size to limit evictions (#17732)

vickenty committed:
Improvements for metrics blocklist (#17592)

* Fix missing build tag

server_test.go and server_bench_test.go depend on things that are only
defined with test tag.

* Add metric exclusion benchmark

* Allow blocking metrics by prefix

* Use binary search for metrics blocklist

* Copy strings before modifying the slice

hannahkm committed:
pkg/config: add additional_endpoints to config_template.yaml (#17672)

Add a section to the config_template.yaml file for the APM additional_endpoints variables. Both the yaml and the environment variables were not documented previously, which can cause problems for documentation.

Co-authored-by: Srdjan Grubor <sgnn7@sgnn7.org>

val06 committed:
java http frameworks client for tests (#17355)

added Java client simulator to be used in e2e tests for java tls feature.

Currently supports:

- native process execution or docker version
- apache, okhttp, httpclient, urlconnection java frameworks

hush-hush committed:
Upgrading FX from v1.18.2 to v1.19.3

This version of FX offers better Hook helpers.

FX changelog:
Add lifecycle hook-convertible methods: StartHook, StopHook,
StartStopHook that can be used with more function signatures.

guyarb committed:
usm: tests: Added timeout to wget implementation (#16905)

nplanel committed:
fix conntrackerTelemetry.unregistersTotal (#17745)

safchain committed:
[CWS] distinguish between snapshot and fallback (#17729)

safchain committed:
[CWS] fix fileless filesystem snapshot (#17715)

blt committed:
Remove the erratic designation from trace-agent RD experiments (#17697)

This commit removes the erratic flag from the two trace-agent Regression
Detector experiments, see #17693 for details. This reverts #17688.

REF SMP-606
REF SMP-330

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>

safchain committed:
[CWS] extract probe monitors (#17742)

tbavelier committed:
[util/containers/filter] Adds registry.k8s.io/pause exclusion (#17753)

safchain committed:
[CWS] fix potential overflow in load module args (#17755)

vboulineau committed:
Fix missing MemoryAccounted in process rt check (#17519)

KevinFairise2 committed:
AP-2159 Remove system-probe and security-agent dependecies from heroku build (#17638)

* Remove system-probe and security-agent deps from keroku

* Update helper function

* Update helper function

* Update postinst script to not crash on heroku flavor

* Run kitchen tests to try on heroku

* Add all builds to test

* Remove all build and all tests

* Run all tests

* Remove variable to run all tests

gh123man committed:
Allow collecting logs by file path with a logs integration config yaml +`ad_identifiers` (#17623)

* support files in logs pod annotations

* reno

safchain committed:
[CWS] improve error output for test timeout (#17730)

Yumasi committed:
inventories: sync USM metadata values with infrastructure-resources (#17739)

Signed-off-by: Guillaume Pagnoux <guillaume.pagnoux@datadoghq.com>

hush-hush committed:
Revert "Upgrading FX from v1.18.2 to v1.19.3"

This reverts commit 9438022e95bc6f7279a440a8c20b3cc94838cfcd.

github-actions[bot] committed:
CWS: sync BTFhub constants (#17777)

Co-authored-by: paulcacheux <paulcacheux@users.noreply.github.com>

dependabot[bot] committed:
Bump golang.org/x/tools from 0.9.3 to 0.10.0 in /pkg/security/secl (#17733)

* Bump golang.org/x/tools from 0.9.3 to 0.10.0 in /pkg/security/secl

Bumps [golang.org/x/tools](https://github.com/golang/tools) from 0.9.3 to 0.10.0.
- [Release notes](https://github.com/golang/tools/releases)
- [Commits](https://github.com/golang/tools/compare/v0.9.3...v0.10.0)

---
updated-dependencies:
- dependency-name: golang.org/x/tools
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Auto-generate go.sum and LICENSE-3rdparty.csv changes

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: dependabot[bot] <dependabot[bot]@users.noreply.github.com>

val06 committed:
fixed prepare_kitchen following directory changes of the tls packages

val06 committed:
fixed prepare_kitchen following directory changes of the tls packages

vboulineau committed:
Inject DD_DOGSTATSD_URL in /config admission controller (#17741)

* Inject DD_DOGSTATSD_URL in /config admission controller

* Update releasenotes-dca/notes/admission-controller-dsd-bf4585681102204f.yaml

Co-authored-by: Ursula Chen <58821586+urseberry@users.noreply.github.com>

---------

Co-authored-by: Ursula Chen <58821586+urseberry@users.noreply.github.com>

remeh committed:
[no-aggregation] return the `MetricSampleBatch` to the pool only once done with it (#17728)

* [no-aggregation] return the `MetricSampleBatch` to the pool only once done with it.

Returning the `MetricSampleBatch` earlier (in the `batcher.go`) was incorrect since
the `NoAggregationStreamWorker` goroutine might still be using it to process its content.

Returning it early was making the MetricSampleBatch available to any running
`TimeSamplerWorker`s, which has the time to write (by indices) its sample in
the MetricSampleBatch, eventually still processed (with access by index) by the
`NoAggregationStreaWorker`, resulting in misuse of the metric sample (or its discard
if it is a no-aggregation unsupported type such as Distribution and Histograms).

AliDatadog committed:
[workloadmeta] Optimize `GetKubernetesPodForContainer`  (#16803)

0intro committed:
[omnibus] Upgrade OpenSCAP to version 1.3.8 (#17763)

This change upgrades OpenSCAP to version 1.3.8.
Consequently, we remove the backported fixes,
along with the changes that were merged upstream.

pgimalac committed:
[Gohai][ASC-571][ASC-549] Implement the new API for the memory package (#17531)

* feat(gohai): implement the new api for the memory package

* fix(gohai/memory): defer close file on Linux

* check actual value in memory test

val06 committed:
fixed python linter

val06 committed:
fixed python linter

jbachorik committed:
Add config to enable Java crash tracking (#17555)

val06 committed:
restored java jar file used by tests

val06 committed:
restored java jar file used by tests

val06 committed:
Merge remote-tracking branch 'origin/valeri.pliskin/restructure-tls-protocols-pkgs' into valeri.pliskin/restructure-tls-protocols-pkgs

# Conflicts:
#	pkg/network/protocols/tls/java/testdata/Wget$1.class
#	pkg/network/protocols/tls/java/testdata/Wget$2.class
#	pkg/network/protocols/tls/java/testdata/Wget.class
